include/hipSYCL/sycl/buffer.hpp

```diff
@@ -600,6 +621,49 @@ class buffer : public detail::property_carrying_object
   : buffer(first, last, AllocatorT(), propList) 
   {}
 
+  template <typename Container,
+            int D = dimensions,
+            typename = std::enable_if_t<D == 1>,
+            typename = detail::EnableIfContiguous<Container, T>>
+  buffer(Container& container, AllocatorT allocator,
+         const property_list& propList = {})
+    : detail::property_carrying_object{propList}
+  {
+    _impl = std::make_shared<detail::buffer_impl>();
+    _alloc = allocator;
+    
+    constexpr bool isConstContainer = std::is_const_v<
+      std::remove_pointer_t<decltype(std::data(container))>>;
+
+    default_policies dpol;
+
+    // If std::data returns non-const pointer, enable write_back
+    if constexpr (isConstContainer)
+      dpol.writes_back = false;
+    else
+      dpol.writes_back = true;
+    
+    dpol.destructor_waits = true;
+    dpol.use_external_storage = true;
```

Comment:
Are we sure we don't want external storage in any case (i.e. never copy input data to internal storage)? For the pointer-based constructors, we:
* Use external storage and write-back for non-const pointers
* Copy to internal storage and disable write-back for const pointers